controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -459,13 +563,28 @@ func (pp *portPublisher) updateEndpoints(endpoints *corev1.Endpoints) {
 	pp.metrics.setExists(true)
 }
 
-func metricLabels(endpoints *corev1.Endpoints) map[string]string {
-	labels := map[string]string{service: endpoints.Name, namespace: endpoints.Namespace}
+func metricLabels(resource interface{}) map[string]string {
+	var serviceName, ns string
+	var resLabels, resAnnotations map[string]string
+	switch res := resource.(type) {
+	case *corev1.Endpoints:
+		{
+			serviceName, ns = res.Name, res.Namespace
+			resLabels, resAnnotations = res.Labels, res.Annotations
+		}
+	case *dv1beta1.EndpointSlice:
+		{
+			serviceName, ns = res.Labels[dv1beta1.LabelServiceName], res.Namespace
```

Comment:
Again, not sure what we need to do in the case when this is not present. I kind of feel this is an invalid scenario and we should just skip processing the slice. Is this the only way to associate an `EndpointSlice` with a service? I assume so. If this label is non present.. then I guess this is an endpoint that is an "orphan" we cannot really do anything with it, so why go through the trouble of processing it at all. Correct me if I am wrong

Reply:
Labels are not the only way of associating an endpointslice to a service. According to the [endpointslice docs](https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/), ownership is established by an ownership reference and/or a label. 

e.g 

```yaml
  ownerReferences:
  - apiVersion: v1
    blockOwnerDeletion: true
    controller: true
    kind: Service
    name: web-svc
    uid: f8d63059-f3ef-4a6f-9bfd-d9742ddb70de
```

There is no resource creation validation happening, so theoretically endpoints can be orphaned if they are created manually without specifying either a label or an ownership ref to the service. While it is unlikely to happen, think there should be a safeguard in place. 

To cope with this, I will add a check to the `addEndpoints` function, a slice that:
* has an addressType != IPv4 (so IPv6 or FQDN)
* or does not have an ownership ref
* or does not have a service label

will not be processed.